### PR TITLE
RAC4720: keep failure environment to help debug

### PR DESCRIPTION
**Pull request by PengTian0**
RAC4720: https://rackhd.atlassian.net/browse/RAC-4720

### CODE REVIEW

File: jobs/FunctionTest/FunctionTest.groovy [hunk 1]
```diff
@@ -85,7 +84,9 @@ def functionTest(String test_name, String label_name, String TEST_GROUP, Boolean
                         "MANIFEST_FILE=${env.MANIFEST_FILE}",
                         "NODE_NAME=${env.NODE_NAME}",
                         "PYTHON_REPOS=ucs-service",
-                        "TEST_TYPE=$test_type"]
+                        "TEST_TYPE=$test_type",
+                        "KEEP_FAILURE_ENV=${env.KEEP_FAILURE_ENV}",
+                        "KEEP_MINS=${env.KEEP_MINS}"]
```
Comment (panpan0000) [inline jobs/FunctionTest/FunctionTest.groovy]: it can be interpreted as keep_minimum or keep_minutes, please make the var name more meaningful.

**Comment by panpan0000** [comment]
there's still ```    trap cleanupVMs SIGINT SIGTERM SIGKILL``` in the code. hmm... but I think it's ok not to touch it,.,.

**Review by panpan0000 — APPROVED** [review]

File: jobs/FunctionTest/FunctionTest.groovy (hunk 1 above)
Comment (PengTian0) [reply jobs/FunctionTest/FunctionTest.groovy]: Thanks.
I changed it to ```KEEP_MINUTES```

**Comment by PengTian0** [comment]
test this please

**Comment by iceiilin** [comment]
test this please

**Comment by PengTian0** [comment]
test this please

File: jobs/FunctionTest/FunctionTest.groovy [hunk 2]
```diff
@@ -140,63 +141,76 @@ def functionTest(String test_name, String label_name, String TEST_GROUP, Boolean
...
+                                    '''
+                                    int failure_count = 0
+                                    int error_count = 0
+                                    if(fileExists ("downstream_file")) {
+                                        def props = readProperties file: "downstream_file"
+                                        failure_count = "${props.failures}".toInteger()
+                                        error_count = "${props.errors}".toInteger()
+                                    }
+                                    if (failure_count == 0 && error_count == 0){
+                                        result = "SUCCESS"
```
Comment (panpan0000) [inline jobs/FunctionTest/FunctionTest.groovy:188]: so if there's stopping error before test.sh runs (like vagrant up failure)
will the failure_count & error_count equal to 0 ? and make result become SUCCESS ?
I'm afraid so ?
Don't merge until this question be addressed.

**Review by panpan0000 — APPROVED** [review]

File: jobs/FunctionTest/FunctionTest.groovy (hunk 2 above)
Comment (PengTian0) [reply jobs/FunctionTest/FunctionTest.groovy:188]: No. if there's stopping error before test.sh runs (like vagrant up failure)
the code under ``` if(exists)``` won't be executed.
So the result is still FAILURE.

**Review by iceiilin — APPROVED** [review]

**Pull request merged**